encoding/ssz/query/buildTree.go

```diff
@@ -0,0 +1,284 @@
+	default:
+		return fmt.Errorf("unsupported SSZ type: %v", info.sszType)
+	}
+}
+
+// addLeafFromBasicType adds a single leaf node for a basic SSZ type.
+func addLeafFromBasicType(t SSZType, v reflect.Value, w *ssz.Wrapper) error {
+	switch t {
+	case Uint8:
+		w.AddUint8(uint8(v.Uint())) // reflect.Uint returns uint64, safe to cast
+	case Uint16:
+		w.AddUint16(uint16(v.Uint()))
+	case Uint32:
+		w.AddUint32(uint32(v.Uint()))
+	case Uint64:
+		w.AddUint64(v.Uint())
+	case Boolean:
+		if v.Bool() {
+			w.AddUint8(1)
+		} else {
+			w.AddUint8(0)
+		}
+	default:
+		return fmt.Errorf("unexpected basic type: %v", t)
+	}
+	return nil
+}
+
+// buildContainerSubtree merkleizes a container by processing each field in order.
+// Fields are padded to the next power of 2 before committing the subtree.
+func buildContainerSubtree(info *SszInfo, v reflect.Value, w *ssz.Wrapper) error {
+	ci, err := info.ContainerInfo()
+	if err != nil {
+		return err
+	}
+
+	// Dereference pointer if needed
+	v = dereferencePointer(v)
+
+	start := w.Indx()
+	for _, name := range ci.order {
+		fieldInfo := ci.fields[name]
+		fieldVal := v.FieldByName(fieldInfo.goFieldName)
+		if err := buildTree(fieldInfo.sszInfo, fieldVal, w); err != nil {
+			return fmt.Errorf("field %s: %w", name, err)
+		}
+	}
+
+	// Container commits require power-of-2 padding
+	addPadding(w, len(ci.order))
+	w.Commit(start)
+
+	return nil
+}
+
+// buildVectorSubtree merkleizes a fixed-length sequence of elements.
+// Basic types are packed into 32-byte chunks; composite types are processed recursively.
+// Elements are padded to the next power of 2 before committing.
+func buildVectorSubtree(info *SszInfo, v reflect.Value, w *ssz.Wrapper) error {
+	vi, err := info.VectorInfo()
+	if err != nil {
+		return err
+	}
+
+	start := w.Indx()
+	length := v.Len()
+
+	if vi.element.sszType.isBasic() {
+		// Vectors of basic types are packed into 32-byte chunks
+		length = packBasicElements(vi.element, v, w, length)
+	} else {
+		// General case: vector of composite elements (each element becomes its own subtree)
+		for i := 0; i < length; i++ {
+			if err := buildTree(vi.element, v.Index(i), w); err != nil {
+				return fmt.Errorf("vector index %d: %w", i, err)
+			}
+		}
+	}
+
+	addPadding(w, length)
+	w.Commit(start)
+	return nil
+}
+
+// buildListSubtree merkleizes a variable-length sequence of elements.
+// Basic types are packed into 32-byte chunks; composite types are processed recursively.
+// The length is mixed into the root hash via CommitWithMixin.
+func buildListSubtree(info *SszInfo, v reflect.Value, w *ssz.Wrapper) error {
+	li, err := info.ListInfo()
+	if err != nil {
+		return err
+	}
+
+	length := v.Len()
+	limit := int(li.Limit())
+	start := w.Indx()
+
+	if li.element.sszType.isBasic() {
+		// Lists of basic types are packed into 32-byte chunks
+		packBasicElements(li.element, v, w, length)
+
+		limit = int(ssz.CalculateLimit(uint64(limit), uint64(length), uint64(itemLength(li.element))))
```

Comment:
```suggestion
	limit := li.Limit()
	start := w.Indx()
	elemInfo := li.element

	if elemInfo.sszType.isBasic() {
		// Lists of basic types are packed into 32-byte chunks
		packBasicElements(elemInfo, v, w, length)

		limit = int(ssz.CalculateLimit(limit, uint64(length), uint64(itemLength(li.element))))
```

Nitpick for reducing type casting & alias for element info